### PR TITLE
Fixing bug for setup FAF

### PR DESCRIPTION
**Pull request by patriczek**
Complete saving url attachement to database
Improved repoassign for rhel

Signed-off-by: Patrik Helia <phelia@redhat.com>

### CODE REVIEW

**Comment by trapas** [comment]
Can one of the admins verify this patch?

**Comment by sorki** [comment]
Should be two commits with proper explanation what each does and why it is required.

File: src/pyfaf/actions/repoassign.py [hunk 1]
```diff
@@ -41,8 +41,15 @@ def run(self, cmdline, db):
         opsys_list = []
 
         for item_name in cmdline.OPSYS + cmdline.ARCH:
+
```
Comment (sorki) [inline src/pyfaf/actions/repoassign.py]: No need for a blank line

File: src/pyfaf/ureport.py [hunk 2]
```diff
@@ -521,6 +521,11 @@ def save_attachment(db, attachment):
         db_url.report = report
         db_url.url = url
         db_url.saved = datetime.datetime.utcnow()
+        
+        try:
+            db.session.flush()
+        except IntegrityError:
+            raise FafError("Unable to save attachment")
```
Comment (sorki) [inline src/pyfaf/ureport.py]: 'Unable to save URL'

File: src/pyfaf/actions/repoassign.py [hunk 3]
```diff
@@ -41,8 +41,15 @@ def run(self, cmdline, db):
...
         for item_name in cmdline.OPSYS + cmdline.ARCH:
+            
             opsys = get_opsys_by_name(db, item_name)
             arch = get_arch_by_name(db, item_name)
+            
+            #If name is rhel we will search Red Hat Enterprise Linux
+            if not (opsys or arch):
+                if item_name == "rhel":
+                    item_name = "Red Hat Enterprise Linux"
+                    opsys = get_opsys_by_name(db, item_name)
```
Comment (sorki) [inline src/pyfaf/actions/repoassign.py]: This doesn't seem right at all. Fixing rhel -> Red Hat Enterprise Linux ambiguity in one place (case) only?

Comment (sorki) [inline src/pyfaf/actions/repoassign.py]: This indeed seems like a only place where it's needed.

So please just move the block bellow the following 'if' as the main condition is the same in both clauses.

**Comment by sorki** [comment]
@trapas add to whitelist

**Pull request merged**